contract/sep-nft-collection.md

```diff
@@ -0,0 +1,124 @@
+## Preamble
+```
+CIP: TBD
+Title: Token Collections (ERC-1155 like)
+Author: Stellar Development Foundation <@stellar>
+Status: Draft
+Discussion: TBD
+Created: 2023-10-13
+Version 0.1.0
+```
+## Summary
+
+This proposal introduces a standard interface for creating and managing NFT collections on Soroban, ensuring compatibility and interoperability within the Stellar ecosystem, and aligning with the previously defined NFT interface and the ERC-1155 standard.
+
+## Motivation
+
+A standardized interface for NFT collections on Soroban will facilitate seamless interaction within the Stellar ecosystem, enhancing user experience and developer interaction by providing a consistent API for NFT collection creation, management, and interaction, while also allowing for efficient batch transfers and management of both fungible and non-fungible tokens within a single contract.
+
+## Abstract
+
+This proposal defines a Rust trait for NFT collections, ensuring that all NFT collection contracts on Soroban adhere to a common API. This API includes functions for creating new NFTs within a collection, transferring NFTs, querying for ownership, and for accessing metadata necessary for displaying NFT information in wallets and other user interfaces.
+
+## Specification
+
+```rust
+pub trait NFTCollectionFactory {
+    // Admin interface – privileged functions.
+    fn initialize(
```

Comment:
Initialization should probably not be a part of the common interface, similar to tokens.